### PR TITLE
Refactor image

### PR DESCRIPTION
**Pull request by JohnDoeMask**
* Encode image to base64 and store it into 'content' field
* Post Model: remove "image_file" , add "content"
* Post Form: Add 'content_type',  'content', and ''unlisted'
* Add javascript for new_post/ to:
- Hide image option when the content type is text/plain, text/markdown,  and application/base64
- Hide content box when the content type is image/png;base64, and image/jpeg;base64

### CODE REVIEW

File: socialdistribution/templates/posts/posts_view.html [hunk 1]
```diff
@@ -19,9 +19,11 @@
...
         <div class="post-item">
             <h5 class="post-card-title">{{ post.title }}</h5>
-            <p class="post-card-text">{{ post.description }}</p>
-            {% if post.image_file != "" %}
-            <p><img src="/media/{{ post.image_file }}"  alt=""></p>
+            {% if post.content_type == "image/png;base64" or post.content_type == "image/jpeg;base64" %}
+            <img src="data:{{ post.content_type }},{{ post.content }}" />
+            {% else %}
+            <p class="post-card-text">{{ post.content }}</p>
+            <!-- <p><img src="/media/{{ post.content }}"  alt=""></p> -->
```
Comment (XiaoleZ) [inline socialdistribution/templates/posts/posts_view.html]: is this a comments? or the code not use. You can delete this line, this you won't use it as reference in the future.

File: socialdistribution/templates/posts/posts_form.html [hunk 2]
```diff
@@ -25,4 +30,22 @@
...
+        $(".fa-camera").show();
+        $("#id_image_file").show();
+      } else {
+        $(".fa-camera").hide();
+        $("#id_image_file").hide();
+        $("#id_content").show();
+      }
+    });
+  });
+</script>
```
Comment (XiaoleZ) [inline socialdistribution/templates/posts/posts_form.html]: I think there is a subfile `static/js` folder. If you could put the javascript in those folder, the html only for html.

Comment (XiaoleZ) [reply socialdistribution/templates/posts/posts_form.html]: add `{% load staticfiles %}` at the top of the template or `{% load static %}`  will work.
For the head section refer to script such as`<script src="{% static "MyApp1/js/scriptForMyApp1.js" %}"></script>`
https://www.reddit.com/r/django/comments/4f7v1n/how_do_i_include_javascript_in_my_django_project/

File: socialdistribution/templates/posts/posts_base.html [hunk 3]
```diff
@@ -19,20 +19,32 @@
         </div>
         <!--end of the post heading-->
         <!--start of the post item-->
+        <script>
```
Comment (XiaoleZ) [inline socialdistribution/templates/posts/posts_base.html]: Add the js to a separate file in `static/js`.

File: socialdistribution/templates/posts/posts_form.html (hunk 2 above)
Comment (JohnDoeMask) [reply socialdistribution/templates/posts/posts_form.html]: Thanks, changes are made

File: socialdistribution/templates/posts/posts_view.html (hunk 1 above)
Comment (JohnDoeMask) [reply socialdistribution/templates/posts/posts_view.html]: Thx!It is removed now

File: socialdistribution/templates/posts/posts_base.html (hunk 3 above)
Comment (JohnDoeMask) [reply socialdistribution/templates/posts/posts_base.html]: Change is made!

File: socialdistribution/templates/posts/posts_form.html [hunk 4]
```diff
@@ -25,4 +29,6 @@
     </div>
   </div>
 </div>
+<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
+<script src="{% static "/js/allthejs.js" %}"></script>
```
Comment (XiaoleZ) [inline socialdistribution/templates/posts/posts_form.html]: why you `load /js/allthejs.js` here?

Comment (JohnDoeMask) [reply socialdistribution/templates/posts/posts_form.html]: Good catch! Thanks Charlotte : )

**Pull request merged**